tabs/utils/urlObject.ts

```diff
@@ -0,0 +1,23 @@
+export const objectFromUrlParams = <T = Record<string | number, any>>(
```

Comment:
Q: `window.create` page should be put in the `tabs`, but not `popup`?

Reply:
I think both are necessary. Metamask distinguishes between tab, window (created using windows.create, considered as background_popup by us), and popup.

1. Metamask uses tab when creating a mnemonic phrase.
2. It uses background_popup when performing connect and swap operations.
3. If we skip background_popup intentionally and click Metamask's popup, it will copy background_popup to this popup.

Not sure if this answers the question.